pkg/operator/ceph/object/controller.go

```diff
@@ -319,3 +340,81 @@ func updateStatus(client client.Client, name types.NamespacedName, status string
 	}
 	logger.Debugf("object store %q status updated to %q", name, status)
 }
+
+func (r *ReconcileCephObjectStore) handleDelete(objectstore *cephv1.CephObjectStore) error {
+	err := r.deleteObjectBucketClaims(objectstore)
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete object bucket claims")
+	}
+
+	err = r.waitForObjectBucketCleanup(objectstore)
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete bject buckets")
+	}
+	return nil
+}
+
+func (r *ReconcileCephObjectStore) deleteObjectBucketClaims(objectstore *cephv1.CephObjectStore) error {
+	objectBucketClaims, err := r.GetObjectBucketClaims(objectstore)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get object bucket claims")
+	}
+
+	for _, obs := range objectBucketClaims.Items {
+		logger.Infof("deleting object bucket claim %q for objectstore %q in namespace %q", obs.Name, objectstore.Name, objectstore.Namespace)
+		err := r.bktclient.ObjectbucketV1alpha1().ObjectBucketClaims("default").Delete(obs.Name, metav1.NewDeleteOptions(0))
+		if err != nil {
+			return errors.Wrapf(err, "failed to delete object bucket claims %q for objectstore %q in namespace %q", obs.Name, objectstore.Name, objectstore.Namespace)
+		}
+	}
+	return nil
+}
+
+func (r *ReconcileCephObjectStore) GetObjectBuckets(objectstore *cephv1.CephObjectStore) (*bktv1alpha1.ObjectBucketList, error) {
+	selector := fmt.Sprintf("bucket-provisioner=%s", bucketProvisioner)
+	objectBuckets, err := r.bktclient.ObjectbucketV1alpha1().ObjectBuckets().List(metav1.ListOptions{LabelSelector: selector})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get buckets for objectstore %q in namespace %q", objectstore.Name, objectstore.Namespace)
+	}
+	return objectBuckets, nil
+}
+
+func (r *ReconcileCephObjectStore) GetObjectBucketClaims(objectstore *cephv1.CephObjectStore) (*bktv1alpha1.ObjectBucketClaimList, error) {
+	selector := fmt.Sprintf("bucket-provisioner=%s", bucketProvisioner)
+	objectBucketsClaims, err := r.bktclient.ObjectbucketV1alpha1().ObjectBucketClaims("default").List(metav1.ListOptions{LabelSelector: selector})
```

Comment:
@travisn @leseb I can't seem to find a way on how not to hard code the `namespace` to get the existing `ObjectBucketClaims`.  Any inputs?

Reply:
I recently introduced a change to watch on the cluster namespace by default. So perhaps you can rely on that since all newly created OBC will work this way.
For backward compatibility, I think hardcoding default is fine though.